Automattic/wp-calypso
# State utils: Convert extendAction to TypeScript

#### Changes proposed in this Pull Request

* Migrates extend action to TypeScript

#### Testing instructions



* Existing unit tests pass
* Permanently delete a comment and ensure that still works as in production. In particular, using redux dev tools, ensure that the `COMMENTS_DELETE` action includes `commentsListQuery` in `meta.comments`
